submissions/usides/js-dom/dist/js/main.js

```diff
@@ -0,0 +1,197 @@
+    portret: 'crim_03',
+    files: ['crim_03_e01', 'crim_03_e02', 'crim_03_e03'],
+  },
+  {
+    number: 123,
+    nick: 'Chill',
+    gender: 'Tom',
+    birth: '01.01.2013',
+    nationality: 'American',
+    marks: 'Fluffy paws',
+    length: 45,
+    weight: 4,
+    color: 'Ginger',
+    details:
+      '- Language: meowing <br>- relaxed all the time <br>- not good at hiding',
+    charges:
+      '- Suspected in tax frauds <br>- Owner of Illegal flip flops market',
+    portret: 'crim_04',
+    files: ['crim_04_e01', 'crim_04_e02'],
+  },
+  {
+    number: 666,
+    nick: 'Katrin',
+    gender: 'Molly',
+    birth: '02.06.2014',
+    nationality: 'Russian',
+    marks: 'Sharp claws',
+    length: 50,
+    weight: 4,
+    color: 'Grey',
+    details:
+      '- Language: meowing, barking <br>- flip flops collector <br>- Charge with extreme caution',
+    charges:
+      '- Suspected in an assault with fangs and claws <br>- Terrorizes all dogs around ',
+    portret: 'crim_05',
+    files: ['crim_05_e01', 'crim_05_e02', 'crim_05_e03'],
+  },
+  {
+    number: 334,
+    nick: 'Sharpei',
+    gender: 'Tom',
+    birth: '01.03.2015',
+    nationality: 'American',
+    marks: 'Super cute',
+    length: 55,
+    weight: 4,
+    color: 'White / grey',
+    details:
+      '- Language: meowing <br>- calls himself as "sharp eye" but for some reason, other calls him as dog breed "shar-pei" <br>- master of disguise as turkey',
+    charges:
+      '- Suspected in mouses assassination with a rifle <br>- mouse killer for money - hitcat',
+    portret: 'crim_06',
+    files: ['crim_06_e01', 'crim_06_e02'],
+  },
+];
+
+const sideNavMenu = document.getElementById('side-nav-menu');
+const sideMenu = document.getElementById('side-nav');
+const menuBtn = document.getElementById('menu-btn');
+const nextBtn = document.getElementById('next-btn');
+let current;
+
+document.addEventListener('DOMContentLoaded', function () {
+  loadMenu();
+  printData(0);
+});
+
+menuBtn.addEventListener('click', function (e) {
+  e.preventDefault();
+  sideMenu.classList.toggle('hidden');
+  this.classList.toggle('active');
+});
+
+nextBtn.addEventListener('click', function (e) {
+  e.preventDefault();
+  current++;
+  printData(current);
+  makeActive(current);
+  if (current == suspects.length - 1) current = -1;
+});
+
+sideNavMenu.addEventListener('click', function (e) {
+  e.preventDefault();
+  if (e.target.nodeName !== 'A') return;
+  window.scrollTo({
+    top: 0,
+    left: 0,
+    behavior: 'smooth',
+  });
+  let arrNum = suspects.findIndex(
+    (item) => item.number === Number(e.target.id.slice(1)),
+  );
+  printData(arrNum);
+  makeActive(arrNum);
+  current = arrNum;
+  sideMenu.classList.add('hidden');
+  menuBtn.classList.remove('active');
+});
+
+function loadMenu() {
+  const fragment = document.createDocumentFragment();
+  for (let i = 0; i < suspects.length; i++) {
```

Comment:
Use a method `map`

Reply:
Could you please explain the reason of `map`? I don't need a modified array here.

Reply:
```
      suspects.map((item, index) => {
          const newElem = document.createElement('li');
          newElem.innerHTML = `<a href="#" class="menu-link" id="p${item.number}">${labelCreate.call(item)}</a>`;
          if(index === 0) {
              newElem.firstElementChild.classList.add('active');
          }
          fragment.appendChild(newElem);
      })
```

Reply:
You can use `Array.prototype.forEach` here as well. But I think your decision with the `for` loop will be Ok for this task.